openhtf/util/logs.py

```diff
@@ -116,10 +119,11 @@ def initialize_record_logger(test_uid, test_record, notify_update):
   logger = get_record_logger_for(test_uid)
   # All record loggers have a shared parent that's separately configured, so
   # we want to propagate to that logger.
-  logger.propagate = True
-  logger.setLevel(logging.DEBUG)
-  # Just in case, make sure we don't have any extra handlers hanging around.
-  logger.handlers = [RecordHandler(test_record, notify_update)]
+  if DEFAULT_TESTRECORD_LEVEL != 'off':
+    logger.propagate = True
+    logger.setLevel(DEFAULT_TESTRECORD_LEVEL.upper())
```

Comment:
I'm supposing that the intention with the logging flags is to filter what appears in the console, but not to filter what ends up in the test record? If that's right then I think the level here needs to be `DEBUG` to ensure all logs are emitted to `test_record.log_records`. We can then filter what goes to `stdout` using the top-level `RECORD_LOGGER`. I got the behavior I wanted by setting the level on the handler instead of the logger itself:

```python
record_handler = logging.StreamHandler(stream=sys.stdout)
record_handler.setLevel(DEFAULT_TESTRECORD_LEVEL.upper())
record_logger = logging.getLogger(RECORD_LOGGER)
record_logger.propagate = False
record_logger.addHandler(record_handler)
```